profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/compute/ProfitBricksComputeServiceAdapter.java

```diff
@@ -0,0 +1,563 @@
+                    String id = String.format("cpu=%d,ram=%s,disk=%f", core, ram, size);
+                    hardwares.add(new HardwareBuilder()
+                            .ids(id)
+                            .ram(ram)
+                            .hypervisor("kvm")
+                            .name(id)
+                            .processor(new Processor(core, 1d))
+                            .volume(new VolumeImpl(size, true, true))
+                            .build());
+                }
+            }
+        }
+        return hardwares;
+    }
+
+    @Override
+    public Iterable<Provisionable> listImages() {
+        // fetch images..
+        ListenableFuture<List<Image>> images = executorService.submit(new Callable<List<Image>>() {
+
+            @Override
+            public List<Image> call() throws Exception {
+                logger.trace("<< fetching images..");
+                // Filter HDD types only, since JClouds doesn't have a concept of "CD-ROM" anyway
+                Iterable<Image> filteredImages = Iterables.filter(api.imageApi().getList(new DepthOptions().depth(1)), new Predicate<Image>() {
+
+                    @Override
+                    public boolean apply(Image image) {
+                        return image.properties().imageType() == Image.Type.HDD;
+                    }
+                });
+                logger.trace(">> images fetched.");
+
+                return ImmutableList.copyOf(filteredImages);
+            }
+
+        });
+        // and snapshots at the same time
+        ListenableFuture<List<Snapshot>> snapshots = executorService.submit(new Callable<List<Snapshot>>() {
+
+            @Override
+            public List<Snapshot> call() throws Exception {
+                logger.trace("<< fetching snapshots");
+                List<Snapshot> remoteSnapshots = api.snapshotApi().list(new DepthOptions().depth(1));
+                logger.trace(">> snapshots feched.");
+
+                return remoteSnapshots;
+            }
+
+        });
+
+        return Iterables.concat(getUnchecked(images), getUnchecked(snapshots));
+    }
+
+    @Override
+    public Provisionable getImage(String id
+    ) {
+        // try search images
+        logger.trace("<< searching for image with id=%s", id);
+        Image image = api.imageApi().getImage(id);
+        if (image != null) {
+            logger.trace(">> found image [%s].", image.properties().name());
+            return image;
+        }
+        // try search snapshots
+        logger.trace("<< not found from images. searching for snapshot with id=%s", id);
+        Snapshot snapshot = api.snapshotApi().get(id);
+        if (snapshot != null) {
+            logger.trace(">> found snapshot [%s]", snapshot.properties().name());
+            return snapshot;
+        }
+        throw new ResourceNotFoundException("No image/snapshot with id '" + id + "' was found");
+    }
+
+    @Override
+    public Iterable<DataCenter> listLocations() {
```

Comment:
Locations are configured via properties, so this method should never be invoked. Let this method just throw an Unsupported op exception.